datafusion/core/src/physical_plan/file_format/parquet/row_groups.rs

```diff
@@ -0,0 +1,744 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::{
+    array::ArrayRef,
+    datatypes::{DataType, Schema},
+};
+use datafusion_common::Column;
+use datafusion_common::ScalarValue;
+use log::debug;
+
+use parquet::{
+    file::{metadata::RowGroupMetaData, statistics::Statistics as ParquetStatistics},
+    schema::types::ColumnDescriptor,
+};
+
+use crate::{
+    datasource::listing::FileRange,
+    physical_optimizer::pruning::{PruningPredicate, PruningStatistics},
+};
+use parquet::basic::{ConvertedType, LogicalType};
+
+use super::ParquetFileMetrics;
+
+pub(crate) fn prune_row_groups(
+    groups: &[RowGroupMetaData],
+    range: Option<FileRange>,
+    predicate: Option<PruningPredicate>,
+    metrics: &ParquetFileMetrics,
+) -> Vec<usize> {
+    // TODO: Columnar pruning
```

Comment:
what here `Columnar pruning` means 🤔

Reply:
I think it means the page index pruning you have implemented -- I 'll remove this comment in a follow on PR